### PR TITLE
bpf-firewall: properly handle kernels where BPF cgroup is disabled bu…

### PR DESCRIPTION
**Pull request by poettering**
…t TRIE maps are enabled

So far, we assumed that kernels where TRIE was on also supported
BPF/cgroup stuff. That's not a correct assumption to make, hence check
for both features separately.

Fixes: #7054

### CODE REVIEW

**Comment by mbiebl** [comment]
Could we cache the results of those checks? It's not like the kernel changes while systemd is running.

**Comment by poettering** [comment]
we do, have a look at the function. "supported" is a static variable and the cache

**Comment by mbiebl** [comment]
> we do, have a look at the function. "supported" is a static variable and the cache

Indeed. Should have checked more carefully. Sorry for the noise.

**Comment by tixxdz** [comment]
lgtm

File: src/core/bpf-firewall.c [hunk 1]
```diff
@@ -676,5 +682,22 @@ int bpf_firewall_supported(void) {
 
         safe_close(fd);
 
+        if (bpf_program_new(BPF_PROG_TYPE_CGROUP_SKB, &program) < 0) {
+                log_debug_errno(r, "Can't allocated CGROUP SKB BPF program, BPF firewalling is not supported: %m");
```
Comment (keszybz) [inline src/core/bpf-firewall.c]: Typo.

**Comment by keszybz** [comment]
Looks good except for the silly typo :]

**Comment by poettering** [comment]
force pushed a new version now with the typo fixed! PTAL!

**Comment by poettering** [comment]
since @tixxdz already liked it, and @keszybz did too, and the only change i made was to a log message string, i took the liberty to set "good-to-merge/waiting-for-ci" on this one

**Comment by poettering** [comment]
and thanks for reviewing!

**Pull request merged**